src/gui/modals/FileDialog.cpp

```diff
@@ -46,44 +46,51 @@ FileDialog::FileDialog( QWidget *parent, const QString &caption,
 
 	setOption( QFileDialog::DontUseNativeDialog );
 
-#ifdef LMMS_BUILD_LINUX
-	QList<QUrl> urls;
-#else
+	auto addIfUnavailable = [](QList<QUrl>& urls, const QString& path)
```

Comment:
I think the name of this lambda is a bit misleading - it sounds like it checks whether the path can't be accessed on the system for some reason. I'd go for something like `addIfNotPresent` or `addUnique` instead.

Also, you could define the lambda after `urls` and capture the variable instead, so you don't need to pass it every time the lambda is called.